src/main/java/com/example/demo/services/ResetPasswordService.java

```diff
@@ -0,0 +1,40 @@
+@Service // Define essa classe como um componente de serviço gerenciado pelo Spring
+public class ResetPasswordService {
+
+    @Autowired // Injeta automaticamente a dependência do UserRepository
+    private UserRepository userRepository;
+
+    // Método responsável por alterar a senha de um usuário
+    public String alterarSenha(String email, String novaSenha, String confirmarSenha) {
```

Comment:
Tu pode definir como parâmetro a classe que tu crio de AlterarSenhaDto, que ele possui esse campos. 
```suggestion
    public String alterarSenha(AlterarSenhaDto dados) {
```
Tu escolhe a variável (o "dados" que for mais fácil mais de identificar o que é que tu tá passando

Reply:
Se fizer a alteração conforme indicado no controller, seguindo contrato do front, ficaria assim:
```suggestion
    public void alterarSenha(AlterarSenhaDto dados) {
```

E não teria os return das strings, teria que dar throw new Exception, alguma exceção, parecido com o que tu fez na linha 12